### PR TITLE
townsquare: profile filter support

### PR DESCRIPTION
**Pull request by thelostone-mc**
##### Description

 - fix bug when tip is not associated with comment / activity
- replace checkbox with My Feed
- enable feed filter for tribe/users with tab param

http://localhost:8000/?tab=my_feed // logged in users feed
http://localhost:8000/?tab=gitcoinco // orgs feed
http://localhost:8000/?tab=danliper // dan's feed

##### Refers/Fixes

This is built on top of https://github.com/gitcoinco/web/pull/6054 by @sounak98

##### Testing

https://share.vidyard.com/watch/1JhpFmhR4bk5UP4jZV1tMW?

### CODE REVIEW

File: app/assets/v2/js/pages/townsquare.js [hunk 1]
```diff
@@ -42,8 +42,9 @@ $(document).ready(function() {
 
   var get_redir_location = function(tab) {
     let trending = $('#trending').is(':checked') ? 1 : 0;
+    let personal = $('#personal').is(':checked') ? 1 : 0;
 
-    return '/?tab=' + tab + '&trending=' + trending;
+    return '/?tab=' + tab + '&trending=' + trending + '&personal=' + personal;
```
Comment (octavioamu) [inline app/assets/v2/js/pages/townsquare.js]: Shouldn't be good to use string literals here?

Comment (sounak98) [reply app/assets/v2/js/pages/townsquare.js]: I think we can remove this as we are making personal feed a separate tab now.

File: app/townsquare/templates/townsquare/index.html [hunk 2]
```diff
@@ -148,7 +148,7 @@ <h5 class="font-weight-semibold font-subheader mb-1">{{nb.title}}</h5>
                     <div class="checkbox_container">
                       <input name="trending" id="trending" type="checkbox" value="0" val-ui='Trending' {% if trending_only %}checked=checked{%endif%} />
                       <span class="checkbox"></span>
-                      <div class="filter-label ml-1">
+                      <div class="filter-label ml-1 mr-2">
```
Comment (sounak98) [inline app/townsquare/templates/townsquare/index.html]: also for the sake of uniformity, let's remove the `mr-2` from here

Comment (thelostone-mc) [reply app/townsquare/templates/townsquare/index.html]: oh nice catch ! Did not see that

**Pull request merged**